File: app/src/main/java/com/example/socialapp/data/repositories/LoginRepository.java
package com.example.socialapp.data.repositories;

import androidx.lifecycle.LiveData;

import com.example.socialapp.models.AccessTokenResponse;
import com.example.socialapp.network.livedataadapter.ApiResponse;

public interface LoginRepository {
    LiveData<ApiResponse<AccessTokenResponse>> getAccessToken(String code,
                                                              String clientId,
                                                              String redirectUri,
                                                              String client_secret);
}

File: app/src/main/java/com/example/socialapp/data/repositories/LoginRepository.kt
package com.example.socialapp.data.repositories

import androidx.lifecycle.LiveData
import com.example.socialapp.models.AccessTokenResponse
import com.example.socialapp.network.livedataadapter.ApiResponse

interface LoginRepository {

    fun getAccessToken(
        code: String,
        clientId: String,
        redirectUri: String,
        clientSecret: String
    ): LiveData<ApiResponse<AccessTokenResponse>>
}
